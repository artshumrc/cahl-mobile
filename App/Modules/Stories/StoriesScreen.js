import React from 'react';
import PropTypes from 'prop-types'
import { ScrollView, View, Text, TextInput, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';
import firebase from 'firebase';
import I18n from 'react-native-i18n';

// components
import Story from './Story.js';

// styles
import styles from './StoriesScreenStyles';

// translations
import en from '../../i18n/languages/english';
import fr from '../../i18n/languages/fr.json';

I18n.translations = {
  en,
  fr,
};

class StoriesScreen extends React.Component {
  constructor(props) {
    super(props);

    this.shareStory = this.shareStory.bind(this);
  }

  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func,
    }).isRequired,
  };

  shareStory() {
    this.props.navigation.navigate('ShareStory')
  }

  render() {
    const { navigation, data, data: { loading, error, stories } } = this.props;

    if (error) {
      console.log(error);
      return (
        <Text style={styles.title}>{`${I18n.t('error')} ${error.message}`}</Text>
      )
    } else if (loading) {
      return (
        <Text style={styles.title}>{I18n.t('loading')}</Text>
      )
    } else {
      return (
        <ScrollView style={styles.container}>
          <View style={styles.headerContainer}>
            <Text style={styles.questionText}>{I18n.t('storiesMessage')}</Text>
            <View style={styles.submitStory}>
              <Text>{I18n.t('shareStory')}</Text>
              <TouchableOpacity onPress={this.shareStory}>
                <Icon name="plus" style={styles.submitIcon} />
              </TouchableOpacity>
            </View>
          </View>
          <View>
            {
              stories && stories.map(story =>
                <Story
                  navigation={navigation}
                  itemId={story._id}
                  key={story._id}
                  content={story.content}
                  userIsOwner={story.userIsOwner}
                  userDisplayName={story.userDisplayName}
                  userPhotoURL={story.userPhotoURL}
                  createdAt={story.createdAt}
                  photoURL={story.photoURL}
                  refetch={data.refetch}
                />
              )
            }
          </View>
        </ScrollView>
      );
    }
  }
}

const getStories = gql`
query getStories {
  stories {
    content,
    userDisplayName,
    userProfilePhotoURL,
    createdAt,
    photoURL,
    _id,
    userIsOwner
  }
}
`;


export default graphql(getStories)(StoriesScreen);
