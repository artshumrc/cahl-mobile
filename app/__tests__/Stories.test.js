import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import StoriesScreen from '../modules/Stories';

it('renders correctly', () => {
  const tree = renderer.create(
    <StoriesScreen />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
