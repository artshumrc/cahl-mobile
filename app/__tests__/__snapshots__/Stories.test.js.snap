// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "paddingBottom": 75,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "flex-start",
            "fontFamily": "AvenirNext-regular",
            "fontSize": 20,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 100,
          }
        }
      >
        What was your personal react to the shootings and media coverage that followed?
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#999999",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "marginBottom": 10,
            "marginLeft": 15,
            "marginRight": 15,
            "marginTop": 10,
          }
        }
      >
        <TextInput
          clearTextOnFocus={true}
          onChange={[Function]}
          style={
            Object {
              "color": "#999999",
              "fontSize": 14,
              "height": 25,
              "marginBottom": 10,
              "marginTop": 10,
              "width": 333,
            }
          }
          value="Share your story"
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#999999",
                  "fontSize": 16,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ÔÅß
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "borderBottomColor": "#999999",
            "borderBottomWidth": 1,
            "paddingBottom": 15,
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
              "marginLeft": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "/Users/tyler/Archimedes/cahl-mobile/App/i/IMG_0487.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 15,
                "height": 30,
                "marginRight": 5,
                "width": 30,
              }
            }
          />
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "AvenirNext-demibold",
                  "fontSize": 14,
                }
              }
            >
              Tyler Kirby
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontFamily": "AvenirNext-regular",
                  "fontSize": 12,
                }
              }
            >
              October 1, 2017
            </Text>
          </View>
        </View>
        <View>
          <Image
            source={
              Object {
                "uri": "/Users/tyler/Archimedes/cahl-mobile/App/images/fabien b 65.png",
              }
            }
            style={
              Object {
                "height": 200,
                "width": 750,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "AvenirNext-regular",
                "fontSize": 14,
                "marginLeft": 15,
                "marginRight": 15,
                "marginTop": 10,
              }
            }
          >
            Minim minim ea consectetur eu deserunt cillum. Consectetur incididunt ad adipisicing nulla aute consequat. Magna reprehenderit ex veniam do esse tempor. Ipsum proident ipsum laborum deserunt nulla enim esse. Proident excepteur tempor anim laborum do aute elit non tempor mollit sunt mollit. Officia irure in qui occaecat est anim commodo anim ut eiusmod adipisicing eu ad est.
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginLeft": 15,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontFamily": "AvenirNext-regular",
                  "fontSize": 14,
                }
              }
            >
              View 3 comments
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#999999",
            "borderBottomWidth": 1,
            "paddingBottom": 15,
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
              "marginLeft": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "/Users/tyler/Archimedes/cahl-mobile/App/images/IMG_0487.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 15,
                "height": 30,
                "marginRight": 5,
                "width": 30,
              }
            }
          />
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "AvenirNext-demibold",
                  "fontSize": 14,
                }
              }
            >
              Tyler Kirby
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontFamily": "AvenirNext-regular",
                  "fontSize": 12,
                }
              }
            >
              October 2, 2017
            </Text>
          </View>
        </View>
        <View>
          <View />
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "AvenirNext-regular",
                "fontSize": 14,
                "marginLeft": 15,
                "marginRight": 15,
                "marginTop": 10,
              }
            }
          >
            Minim minim ea consectetur eu deserunt cillum. Consectetur incididunt ad adipisicing nulla aute consequat. Magna reprehenderit ex veniam do esse tempor. Ipsum proident ipsum laborum deserunt nulla enim esse. Proident excepteur tempor anim laborum do aute elit non tempor mollit sunt mollit. Officia irure in qui occaecat est anim commodo anim ut eiusmod adipisicing eu ad est.
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginLeft": 15,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#999999",
                  "fontFamily": "AvenirNext-regular",
                  "fontSize": 14,
                }
              }
            >
              View 2 comments
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
